### PR TITLE
Makes request headers configurable

### PR DESCRIPTION
**Pull request by adatta02**
Addresses the issue in https://github.com/seantomburke/sitemapper/issues/49

Use with:

```
const sm = new Sitemapper({
  url: 'https://art-works.community/sitemap.xml',
  timeout: 15000, // 15 seconds,
  requestHeaders: {
    'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:81.0) Gecko/20100101 Firefox/81.0'
  }
});

sm.fetch()
  .then(data => console.log(data.sites))
  .catch(error => console.log(error));
```

### CODE REVIEW

File: src/assets/sitemapper.js [hunk 1]
```diff
@@ -150,12 +155,13 @@ export default class Sitemapper {
         // The promise resolved, remove the timeout
         clearTimeout(this.timeoutTable[url]);
 
+        console.log(error);
```
Comment (seantomburke) [inline src/assets/sitemapper.js:158]: Is this intentional to log the error or just left over? If intentional you can use console.error(error)

**Review by seantomburke — APPROVED** [review]

File: src/assets/sitemapper.js (hunk 1 above)
Comment (adatta02) [reply src/assets/sitemapper.js]: Whoops, unintentional. Just removed.

**Review by seantomburke — APPROVED** [review]
Looks good to me

**Comment by adatta02** [comment]
@seantomburke looks like http://wp.seantburke.com/sitemap.xml is MIA which is making the CI fail

**Comment by seantomburke** [comment]
> @seantomburke looks like http://wp.seantburke.com/sitemap.xml is MIA which is making the CI fail

@adatta02 Looks like it needed to be updated to https://wp.seantburke.com, and also cnn.com's sitemap.xml moved. Updated in new PR #51

File: src/assets/sitemapper.js [hunk 2]
```diff
@@ -109,7 +111,10 @@ export default class Sitemapper {
           return xmlParse(response.body);
         })
         .then(data => resolve({ error: null, data }))
-        .catch(response => resolve({ error: response.error, data: {} }));
+        .catch(response => {
+          console.log(response);
```
Comment (seantomburke) [inline src/assets/sitemapper.js:115]: Just revert this part, rebase to have tests working again, and should be good to merge

**Comment by seantomburke** [comment]
Merged with #56

**Pull request closed without merging**